airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresCatalogHelper.java

```diff
@@ -61,6 +58,42 @@ public static AirbyteStream overrideSyncModes(final AirbyteStream stream) {
     return stream.withSupportedSyncModes(Lists.newArrayList(SyncMode.FULL_REFRESH, SyncMode.INCREMENTAL));
   }
 
+  public static List<String> getViewsInSchema(final JdbcDatabase database, final String schema) throws SQLException {
+    final String query = "SELECT viewname FROM pg_catalog.pg_views WHERE schemaname = ?";
+    final List<JsonNode> result = database.queryJsons(query, schema);
+    return result.stream()
+        .map(jsonNode -> jsonNode.get("viewname").asText())
+        .collect(Collectors.toList());
+  }
+
+  public static Map<String, List<String>> getViewsForAllSchemas(final JdbcDatabase database, final List<String> schemas) throws SQLException {
+    Map<String, List<String>> viewsBySchema = new HashMap<>();
+    for (String schema : schemas) {
+      List<String> views = getViewsInSchema(database, schema);
+      viewsBySchema.put(schema, views);
+    }
+    return viewsBySchema;
+  }
+
+  public static boolean isStreamAView(Map<String, List<String>> viewsBySchema, final AirbyteStream stream) {
+    return viewsBySchema.getOrDefault(stream.getNamespace(), Collections.emptyList()).contains(stream.getName());
+  }
+
+  /**
+   * This method is used for CDC sync in order to overwrite sync modes for cursor fields cause cdc use
+   * another cursor logic. Used in discovery when xmin is configured.
+   *
+   * @param stream - airbyte stream
+   * @param viewsBySchema - view names by schema
+   * @return will return list of sync modes where view streams will have only FULL_REFRESH sync mode
+   */
+  public static AirbyteStream overrideSyncModes(final AirbyteStream stream, Map<String, List<String>> viewsBySchema) {
+    if (isStreamAView(viewsBySchema, stream)) {
+      return stream.withSupportedSyncModes(Lists.newArrayList(SyncMode.FULL_REFRESH));
```

Comment:
I'm not sure, but I thought in an xmin sync, full refreshes were still incremental... and the therefore still dependent on a usable cursor?

Reply:
I think this is for the cursor-based mode for xmin sync; 

maybe I'm wrong, but from what I tested, it does not require specifying a cursor in xmin  (see picture below)
![Screenshot 2024-07-18 at 7 30 15 PM](https://github.com/user-attachments/assets/8043407b-ab5b-4b3e-a0d4-8d7d65091082)

Reply:
The full refreshes are still incremental (based on CTID), but I didn't think a cursor needed to be explicitly set? So I think fixing the sync modes might work 

cc : @xiaohansong who might know this

Reply:
yes - cursor is not needed for full refresh. We should support full refresh anyway, for views they would be non resumable.